src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -265,4 +265,16 @@ ReadBufferFromRemoteFSGather::~ReadBufferFromRemoteFSGather()
         appendUncachedReadInfo();
 }
 
+bool ReadBufferFromRemoteFSGather::isSeekCheap()
+{
+    return !current_buf || current_buf->isSeekCheap();
+}
+
+bool ReadBufferFromRemoteFSGather::isContentCached(size_t offset, size_t size)
+{
+    if (!current_buf)
+        initialize();
+
+    return current_buf && current_buf->isContentCached(offset, size);
```

Comment:
I suspect we should adjust offset here in the same way we do it in `initialise`:

https://github.com/ClickHouse/ClickHouse/blob/c52571223e408779c21f911dd6d4ff08271f80ed/src/Disks/IO/ReadBufferFromRemoteFSGather.cpp#L124-L146

but there is only one blob for each object for MT tables. so what would be a good way to reproduce a case when there are multiple objects?

Reply:
> so what would be a good way to reproduce a case when there are multiple objects?

see `tests/integration/test_log_family_s3`.
Log family engines use append operation in disks, but for s3 we implement append by adding a new file on s3 instead of rewriting the whole file on each append.
Also if you enable experimental transactions for merge tree, then they also do append to (if I remember correctly) `format_version.txt` file, so more there will also be more than one s3 file for a single clickhouse file.